src/LibraryUtilities.ts

```diff
@@ -250,6 +286,34 @@ export function constructLibraryItem(
 }
 
 /**
+ * Compare two arrays of strings and check if the strings in the first array match
+ * those in the second array.
+ * 
+ * @param {string[]} parts 
```

Comment:
Can we use names like `firstParts` and `secondParts` to make the method more generic? For `returns` documentation, if you can provide few examples and their corresponding return values? Can we return `-1` if no match is found instead of `undefined`?